keps/sig-instrumentation/4200-healthcheck-metrics/README.md

```diff
@@ -0,0 +1,297 @@
+
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+Via feature gate.
+
+
+###### Does enabling the feature change any default behavior?
+
+Yes, it exposes an endpoint which can then be scraped.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, but it will remove the endpoint.
+
+### Monitoring Requirements
+
+I am proposing a metric/series of metrics here.
+
+### Scalability
+
+We already hit the readyz/healthz/livez endpoints of our control-plane components frequently, this KEP only 
+adds instrumentation of these endpoints' results.
+
+###### Will enabling / using this feature result in any new API calls?
+
+Yes, we are proposing that this health metrics are surfaced in each component under `/metrics/health` which 
+will have to be consumed for the feature to be useful. However, this should be relatively innocuous since 
+it will an isolated endpoint strictly for the purpose of surfacing health metrics.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No. 
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No. 
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No. This should, in theory, reduce calls to health endpoints since SLIs need to be calculated currently by directly hitting
+and parsing out the results of our existing health check endpoints, which adds to the total number of calls (since kubelet 
+also hits these endpoints). 
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No/
+
+### Troubleshooting
+
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+If apiserver is unavailable, then you will not be able to ingest the metrics from the apiserver.
+However, the failure of etcd should allow you to scrape the metrics from apiserver, so long as 
+it is otherwise healthy. 
+
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+## Drawbacks
+
+Slight increase of memory usage for components (i.e. the breadth of the prometheus metric label values).
+
+## Alternatives
```

Comment:
can not be used the same endpoint and discriminate with a header?

something similar to `"Accept: application/vnd.kubernetes.protobuf"`  but with a different application name

Reply:
That would be extremely strange behavior to toggle metrics using a header.